builtin/credential/ldap/path_config.go

```diff
@@ -143,8 +166,22 @@ func (b *backend) pathConfigWrite(
 	if groupdn != "" {
 		cfg.GroupDN = groupdn
 	}
+	groupfilter := d.Get("groupfilter").(string)
```

Comment:
I ended up generalizing the default loading. Have a look at `backend.Config`, `backend.newConfigEntry`, `backend.pathConfigRead`, `backend.pathConfigWrite`. I believe this will do the right thing for upgraders, but keep me honest.

I also added a test: `TestBackend_configDefaultsAfterUpdate`